### PR TITLE
Core/Spells: Move spell anchor here to generic because is used by mor…

### PR DESCRIPTION
**Pull request by aquadeus**
…e bosses

**Changes proposed:**

-  Move script of spell "Anchor here" to generic spell

**Tests performed:**

Tested in-game

### CODE REVIEW

File: sql/updates/world/master/2022_99_99_99_world_anchor_here.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+DELETE FROM `spell_script_names` WHERE `spell_id` = 45313 AND `ScriptName` = 'spell_sc_anchor_here';
+INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) VALUES (45313, 'spell_anchor_here');
```
Comment (mdX7) [inline sql/updates/world/master/2022_99_99_99_world_anchor_here.sql:2]:
```sql
DELETE FROM `spell_script_names` WHERE `ScriptName`='spell_anchor_here';
INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`)
    SELECT `ssn`.`spell_id`, 'spell_anchor_here'
	FROM `spell_script_names` AS `ssn`
	WHERE `ScriptName`='spell_sc_anchor_here';
DELETE FROM `spell_script_names` WHERE `ScriptName`='spell_sc_anchor_here';
```

Comment (offl) [reply sql/updates/world/master/2022_99_99_99_world_anchor_here.sql:2]: Why not just
```sql
UPDATE `spell_script_names` SET `ScriptName` = 'spell_gen_anchor_here' WHERE `ScriptName` = 'spell_sc_anchor_here';

Comment (mdX7) [reply sql/updates/world/master/2022_99_99_99_world_anchor_here.sql:2]: agreed xD

**Comment by offl** [comment]
That 'gen' prefix is to avoid potential collision between script names

File: src/server/scripts/Spells/spell_generic.cpp [hunk 2]
```diff
@@ -4732,6 +4732,23 @@ class spell_summon_battle_pet : public SpellScript
...
 };
 
+// 45313 - Anchor Here
+class spell_gen_anchor_here : public SpellScript
+{
+    PrepareSpellScript(spell_gen_anchor_here);
+
+    void HandleScript(SpellEffIndex /*effIndex*/)
+    {
+        if (Creature* creature = GetHitUnit()->ToCreature())
```
Comment (Shauren) [inline src/server/scripts/Spells/spell_generic.cpp:4742]:
```suggestion
        if (Creature* creature = GetHitCreature())
```

**Pull request merged**